frontend/src/components/Header/StopClock.js

```diff
@@ -60,31 +59,31 @@ class StopClock extends Component<Props, State> {
     currentTime: Date.now(),
   };
 
-  tick = () => {
-    this.setState({
-      currentTime: Date.now(),
-    });
-  };
-
-  timer = null;
-  endTime = 0;
   componentDidMount() {
-    this.timer = setInterval(this.tick, 500);
     this.endTime = this.props.startTime + 1000 * this.props.countdown;
   }
 
   componentDidUpdate() {
     if (this.endTime !== 0 && Date.now() >= this.endTime) {
       clearInterval(this.timer);
       this.endTime = 0;
-      this.props.history.push('/afterward');
+      this.props.onTimeout();
     }
   }
 
   componentWillUnmount() {
     clearInterval(this.timer);
   }
 
+  timer: IntervalID = setInterval(this.tick, 500);
```

Comment:
Why did you move this out of `componentDidMount`? That's the best place for setting up timers.

Reply:
@jkimbo It was kind of forced by Flow. IntervalID should only be an int or null (ever) but flow doesn't allow null or int, only the IntervalID type which is returned from a setInterval. So you have to initialise the field with a dummy setInterval i.e. `timer: IntervalID = setInterval(() => {}, 500)` ... but then in this case I could've just initialised with the real tick method
Happy to use the dummy interval or eslint-disable on this line if you think that's better

Reply:
yeah that works :)